### PR TITLE
[MAINTENANCE] Unit tests for `CheckpointStore`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Unit tests for `CheckpointStore`

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/checkpoint/checkpoint.py [hunk 1]
```diff
@@ -73,8 +73,11 @@ def __init__(
         checkpoint_config: CheckpointConfig,
         data_context: "DataContext",  # noqa: F821
     ) -> None:
-        # Note the gross typechecking to avoid a circular import
-        if "DataContext" not in str(type(data_context)):
+        from great_expectations.data_context.data_context.abstract_data_context import (
+            AbstractDataContext,
+        )
+
+        if not isinstance(data_context, AbstractDataContext):
```
Comment (cdkini) [inline great_expectations/checkpoint/checkpoint.py:80]: Better error check using an inline import - also makes this more testable.

File: tests/data_context/store/test_checkpoint_store.py [hunk 2]
```diff
@@ -22,6 +30,15 @@
 logger = logging.getLogger(__name__)
 
 
+@pytest.fixture
+def checkpoint_store_with_mock_backend() -> Tuple[CheckpointStore, mock.MagicMock]:
```
Comment (anthonyburdi) [inline tests/data_context/store/test_checkpoint_store.py:34]: I like it!

File: tests/data_context/store/test_checkpoint_store.py [hunk 3]
```diff
@@ -350,3 +367,164 @@ def test_serialization_self_check(capsys) -> None:
...
+
+
+@pytest.mark.unit
+def test_get_checkpoint_with_invalid_legacy_checkpoint_raises_error(
+    checkpoint_store_with_mock_backend: Tuple[CheckpointStore, mock.MagicMock]
+) -> None:
+    store, mock_backend = checkpoint_store_with_mock_backend
+    mock_backend.get.return_value = (
+        CheckpointConfig().to_json_dict()
+    )  # Defaults to empty LegacyCheckpoint
```
Comment (anthonyburdi) [inline tests/data_context/store/test_checkpoint_store.py:504]: Thanks for the comment 👍

File: tests/data_context/store/test_checkpoint_store.py [hunk 4]
```diff
@@ -350,3 +367,164 @@ def test_serialization_self_check(capsys) -> None:
...
+        store.get_checkpoint(name="my_checkpoint", ge_cloud_id=None)
+
+    assert (
+        "Attempt to instantiate LegacyCheckpoint with insufficient and/or incorrect arguments"
+        in str(e.value)
+    )
+
+
+@pytest.mark.unit
+def test_add_checkpoint(
```
Comment (anthonyburdi) [inline tests/data_context/store/test_checkpoint_store.py:516]: Does it make sense to call this a legacy checkpoint test and create a separate test with non legacy checkpoint?

**Review by anthonyburdi — COMMENTED** [review]
Just a question to address before approval.

File: tests/data_context/store/test_checkpoint_store.py (hunk 4 above)
Comment (cdkini) [reply tests/data_context/store/test_checkpoint_store.py:516]: so this particular test doesn't actually have any logical around legacy vs current checkpoints (while the one above did). For that reason, I don't think we need to get as granular and separate it out with more tests.

Let me know if you disagree and I'd be happy to add another test.

**Review by anthonyburdi — APPROVED** [review]

File: tests/data_context/store/test_checkpoint_store.py (hunk 4 above)
Comment (anthonyburdi) [reply tests/data_context/store/test_checkpoint_store.py:516]: I think that makes sense. You would only be testing that add_checkpoint was called with a different checkpoint config which shouldn't be different than the test as written.

**Pull request merged**